docs/source/daml2ts/index.rst

```diff
@@ -14,62 +14,87 @@ Introduction
+The code ``daml2ts`` generates uses the library `@daml/types <https://github.com/digital-asset/daml/tree/master/language-support/ts/daml-types>`_.
 
-Primitives: @daml/types
------------------------
+Usage
+-----
 
-To understand the TypeScript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the primitives provided by @daml/types.
+``daml2ts`` is invoked via the DAML SDK assistant.
 
-**Interfaces**:
+In outline, the command to generate TypeScript from DAML is ``daml codegen ts DAR -o OUTDIR`` where ``DAR`` is the path to a DAR file (generated via ``daml build``) and ``OUTDIR`` is a directory where you want the TypeScript to be written.
+
+Here's a complete example that generates TypeScript from a project produced from the standard "skeleton" template.
+
+.. code-block:: bash
+   :linenos:
+
+   daml new my-proj skeleton # Create a new project based off the skeleton template
+   cd my-proj # Enter the newly created project directory
+   daml build  # Compile the project's DAML files into a DAR
+   daml codegen ts .daml/dist/my-proj-0.0.1.dar -o generated/ts # Generate Typescript from the DAR
 
-- ``Serializable<T>``
-- ``Template<T extends object, K = unknown>``
-- ``Choice<T extends object, C, R, K = unknown>``
+- On execution of these commands:
 
-  ..
-        An object that extends the ``Serializable<T>`` interface has a field ``decoder: () => jtv.Decoder<T>``.
-        An object that extends the ``Template<T, K>`` interface has fields ``templateId: string``, ``keyDecoder: ()=> jtv.Decoder<K>``, ``Archive: Choice<T, {}, {}>`` and extends ``Serializable<T>``.
-        An object that extends the ``Choice<T, C, R, K>`` interface has fields ``template: () => Template<T, K>``, ``argumentDecoder: () => jtv.Decoder<C>``, ``resultDecoder: () => jtv.Decoder<R>`` and ``choiceName: string``.
+  - The directory ``my-proj/generated/ts`` contains the generated TypeScript source files;
+  - The files are arranged into directories - each directory having a numeric name;
+  - The primary directory contains the TypeScript corresponding to the DAML files in ``my-proj/daml``, the remaining directories contain supporting TypeScript corresponding to modules of the DAML standard library;
+  - For example, ``generated/ts/03d1a767741153b5910ad849b0f7af96d205416b3e0ec4c23cbc75a0177776f4/Main.ts`` contains the TypeScript definitions for ``daml/Main.daml``.
+
+To get a quickstart idea of how to use what has been generated, you may wish to jump to the `Templates and choices`_ section and return to the reference material that follows as needed.
+
+Primitive DAML types: @daml/types
+---------------------------------
+
+To understand the TypeScript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the TypeScript equivalents of the primitive DAML types provided by @daml/types.
+
+**Interfaces**:
+
+- ``Template<τ extends object, κ = unknown>``
```

Comment:
We should use `T` and `K` here because they are variables in the object languages, i.e., type variables in TypeScript.

Reply:
✔️